crates/table/src/page.rs

```diff
@@ -140,7 +146,9 @@ struct FixedHeader {
 
     // TODO(stable-module-abi): should this be inlined into the page?
     /// For each fixed-length row slot, true if a row is stored there,
-    /// false if the slot is uninit.
+    /// false if the slot is unallocated.
+    ///
+    /// Unallocated row slots store valid-unconstrained bytes, i.e. are never `poison`.
```

Comment:
I disagree with the choice Rust has made here; I think `poison` is a much better description. In particular, I think it's very non-obvious based on the name `uninit` why `let mut x = [0; N]; x.copy_from_slice(something);` can leave `x` containing uninit, but `poison` is much more obvious.

Basically, "uninit" implies that it's a state that a byte of memory can leave but never re-enter, whereas "poison" makes clear that it's an invalid value that a byte can take.

Reply:
> There is a semantic difference between LLVM and Rust here which explain why "poison" is not used. If x = [0; N] then x.copy_from_slice(something) for a valid something can never make x contain uninit, as the undefined behavior would occur before reaching the call simply by materializing &[u8] pointing to uninit bytes.

Well, the initial `[0; N]` can never contain uninit/poison either iff it's ascribed a type which doesn't permit poison. That's just because the type system excludes it. If we had overloadable literals or I wrote `[MaybeUninit::new(0); N]`, Rust doesn't magically cause the `copy_from_slice`/`memcpy` to not poison the destination.

> I think this is a settled debate though, and even if we may disagree, we should use the standard terminology of Rust.

I'll do it, but I'm not happy about it.